contracts/gatekeepers/ContributionIncrementGateKeeper.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+import "./IGateKeeper.sol";
+
+/// @notice A gateKeeper that limits the amount that can be contributed per contribution.
+contract ContributionIncrementGateKeeper is IGateKeeper {
+    uint96 private _lastId;
+
+    struct ContributionIncrement {
+        uint96 min;
+        uint96 max;
+    }
+
+    /// @notice Get the merkle root used by a gate identifyied by it's `id`.
+    mapping(uint96 => ContributionIncrement) public contributionLimits;
+
+    /// @inheritdoc IGateKeeper
+    function isAllowed(address, bytes12 id, bytes memory userData) external view returns (bool) {
+        uint96 amount = abi.decode(userData, (uint96));
```

Comment:
Isn't `userData` user-supplied? They can just fib. I think we need to modify `Crowdfund` to pass `msg.value` into `isAllowed()`?

Reply:
Oof, yeah I'll fix this